build.sh

```diff
@@ -49,6 +50,10 @@ echo '::endgroup::'
 echo '::group::Publishing'
 git remote add aur "ssh://aur@aur.archlinux.org/${pkgname}.git"
 git add -fv PKGBUILD .SRCINFO
-git commit --allow-empty -m "$commit_message"
-git push -fv aur master
+if [[ $allow_empty_commits == 'true' ]]; then
+    git commit --allow-empty -m "$commit_message"
+else
+    git diff-index --quiet HEAD || git commit -m "$commit_message" # use `git diff-index --quiet HEAD ||` to avoid error
+fi
```

Comment:
Since `$allow_empty_commits` is either `true` or `false`, I prefer a `case...in...esac` syntax.

```sh
case "$allow_empty_commits" in
true) action 1;;
false) action 2;;
*) error and exit;;
esac
```